### PR TITLE
Fixed deepmerge module export naming issue

### LINKED ISSUE #899: deepmerge needs to be updated to latest version to prevent webpack bug
#### Issue Summary

The current `deepmerge` version doesn't work with Webpack:

```
TypeError: merge is not a function
    at Personalization.deepMergeDynamicTemplateData (/opt/sulvo/cloudflare-app/handlers/.webpack/service/webpack:/node_modules/@sendgrid/helpers/classes/personalization.js:289:1)
...
```

This is a known [issue](https://github.com/TehShrike/deepmerge/issues/137).

#### Steps to Reproduce

1. Setup Webpack
2. Send a transactional email injecting dynamic template data, e.g:

```javascript
import sgMail from '@sendgrid/mail'

sgMail.setApiKey(process.env.SENDGRID_API_KEY)

sgMail
  .send({
    to: 'me@sendgrid.com',
    from: 'no-reply@sendgrid.com',
    templateId: '...',
    dynamicTemplateData: {
      name: '',
    },
  }).catch(console.error)
```

#### Technical details:

* sendgrid-nodejs version: 6.3.1
* Node.js version: 10.15.1

### PR DESCRIPTION
**Pull request by chrisamador**
Fixes #899  

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the development branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
This PR fixes webpack's interpolation issue with the deepmerge package. Related to the changes made to [deepmerge](https://github.com/TehShrike/deepmerge/pull/124). Renaming `merge` to `deepmerge` fixes the issue.

### CODE REVIEW

**Comment by pukhalski** [comment]
Can it be merged please?

File: packages/helpers/classes/personalization.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const EmailAddress = require('./email-address');
 const toCamelCase = require('../helpers/to-camel-case');
 const toSnakeCase = require('../helpers/to-snake-case');
 const deepClone = require('../helpers/deep-clone');
-const merge = require('deepmerge');
+const deepmerge = require('deepmerge');
```
Comment (thinkingserious) [inline packages/helpers/classes/personalization.js]: For consistency, how about `deepMerge`?

**Comment by adamreisnz** [comment]
Interesting bug, seems like this is something that should be fixed in Webpack ideally, as we should be free to call our variables however we want, without it affecting the build.

But happy with this change if it fixes the issue for the time being, either `deepmerge` if that's what's necessary to fix it, or otherwise `deepMerge` as @thinkingserious suggested 👍

**Comment by thinkingserious** [comment]
Thanks for weighing in Adam!

**Comment by thinkingserious** [comment]
Hello @chrisamador,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by CharlBest** [comment]
Hi there.

Is there any chance that this package can be deployed more often and not every 6 months or so as we need some of the bugs fixes. For example my mails are broken because of this issue. Unfortunately I can't downgrade Webpack because I'm using the Angular CLI which has Webpack as a peer dependency which means I have to use the version that they provide. So I'm stuck now and based on the release schedule for this project I'm still going to wait a long time is that right?

Thanks for fixing these issues.

**Comment by richardcase** [comment]
I'm in the same situation and would really a appreciate a new release with this fix in.

**Comment by adamf** [comment]
Hi! I'm also hitting this bug, and it's preventing us from using sendgrid. Any eta or workaround?

**Comment by aymericbouzy** [comment]
@thinkingserious would it be possible to publish a new version with this fix included? It's blocking us as well.

**Comment by childish-sambino** [comment]
Releasing soon ...

**Pull request merged**